app/javascript/packs/CuratedClustersButton.jsx

```diff
@@ -0,0 +1,23 @@
+import { h, render } from 'preact';
+import { CuratedClusters } from '../articles/components/CuratedClustersButton';
+
+function loadElement() {
+  const root = document.getElementById('Curated-Collections-Button');
+  if(root) {
+    render(
+      <CuratedClusters 
+        article_id={root.dataset.article}
+        user_id={root.dataset.userid}
+        clusters = {root.dataset.clusters}
+      />,
+      root,
+      root.firstElementChild,
+    );
+  }
+}
```

Comment:
no idea what this does but it looks great!

Reply:
This is part of the pack, it does the initial rendering of our CuratedClusters component. The typical pattern in this project is erb =>pack => component

Reply:
thats super helpful!